docs/source/conf.py

```diff
@@ -0,0 +1,99 @@
+"""Configuration file for the Sphinx documentation builder.
+
+This file only contains a selection of the most common options. For a full
+list see the documentation:
+https://www.sphinx-doc.org/en/master/usage/configuration.html
+"""
+
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+# pylint: disable=W,C,R
+import os
+import sys
+
+from pathlib import Path
+from sphinx.ext.apidoc import main
+
+try:
+    import toml
+except ImportError:
+    sys.exit("Please make sure to `pip install toml` or enable the Poetry shell and run `poetry install`.")
+
+# -- Variable setup --------------------------------------------------------------
+
+ROOT_DIR = Path(__file__).parent.parent.parent
+CURR_DIR = f"{ROOT_DIR}/docs/source"
+PYPROJECT_CONFIG = toml.load(f"{ROOT_DIR}/pyproject.toml")
+TOOL_CONFIG = PYPROJECT_CONFIG["tool"]["poetry"]
+
+# Inserts the diffsync library into the path. This is needed for RTD env to find the
+# library needed for autodocs.
+sys.path.insert(0, os.path.abspath("../.."))
+
+# -- Project information -----------------------------------------------------
+
+project = TOOL_CONFIG["name"]
+copyright = f"2021, {','.join(TOOL_CONFIG['authors'])}"
+author = ",".join(TOOL_CONFIG["authors"])
+
+# The full version, including alpha/beta/rc tags
+release = TOOL_CONFIG["version"]
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = ["sphinx.ext.autodoc", "sphinx.ext.napoleon", "m2r2"]
```

Comment:
I'm familiar with `autodoc` and `napoleon` - for my information, what is `m2r2` and what does it provide?